gcsio/src/test/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageGrpcReadChannelTest.java

```diff
@@ -1176,6 +1101,8 @@ public void testReadCachedFooterPartiallyWithInplaceSeek() throws Exception {
     readChannel.position(readOffset);
     readChannel.read(buffer);
```

Comment:
With reading 13MB-15MB offset, why do we request for data till end of file ?

With position to 13MB call, should we change to FADVISE.RANDOM and read only 2MB ?

Reply:
> With reading 13MB-15MB offset, why do we request for data till end of file ?

`getBytesToRead` returns `empty` if footer is not read and FADVISE.AUTO. We don't actually read till EOF, just that we don't set `readLimit`.

> With position to 13MB call, should we change to FADVISE.RANDOM and read only 2MB ?

We don't have sufficient information to change the behaviour. All we know is that it may be a sequential read from a non-zero offset.